engine/gctscript.go

```diff
@@ -0,0 +1,82 @@
+package engine
+
+import (
+	"fmt"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/thrasher-corp/gocryptotrader/gctscript/vm"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+const gctscriptManagerName = "GCTScript"
+
+type gctScriptManager struct {
+	running  atomic.Value
+	shutdown chan struct{}
+}
+
+// Started returns if gctscript manager subsystem is started
+func (g *gctScriptManager) Started() bool {
+	return g.running.Load() == true
+}
+
+// Start starts gctscript subsystem and creates shutdown channel
+func (g *gctScriptManager) Start() error {
+	if g.Started() {
+		return fmt.Errorf("%s %s", gctscriptManagerName, ErrSubSystemAlreadyStarted)
+	}
+	log.Debugln(log.Global, gctscriptManagerName, MsgSubSystemStarting)
+	g.shutdown = make(chan struct{})
+	go g.run()
+	return nil
+}
+
+// Stop stops gctscript subsystem along with all running Virtual Machines
+func (g *gctScriptManager) Stop() error {
+	if !g.Started() {
+		return fmt.Errorf("%s %s", gctscriptManagerName, ErrSubSystemAlreadyStopped)
+	}
+
+	log.Debugln(log.Global, gctscriptManagerName, MsgSubSystemShuttingDown)
```

Comment:
https://github.com/thrasher-corp/gocryptotrader/pull/383/files#diff-9312417691ccc0ddaba643d50097ba0cR434 Same for this guy